### DISCUSSION TITLE
LDPC5GDecoder

### DISCUSSION THREAD

**Discussion by zhiha** [NVlabs/sionna#120] [General]
When I simulate with Sionna, the current k=3800 and 1/2 coding is used. However, I find that the speed of LDPC5GDecoder decoding is too slow. Am I using it wrong? Is there any method to speed up the decoding process?

**Comment by SebastianCa** [comment]
Hi,

there are multiple ways how you can speed-up your simulations:
- Please ensure that you run the code in Graph/XLA mode (see [Eager vs. Graph Mode](https://nvlabs.github.io/sionna/examples/Sionna_tutorial_part1.html#Eager-vs-Graph-Mode))
- Increase the batch-size to benefit from "inter-frame" parallelization, i.e., the simultaneous processing of multiple independent codewords. A small batch-size will most likely result in a low GPU load
- Please note that the decoding speed also strongly depends on your hardware. Do you have GPU support activated? In Google Colab you need to select the runtime first.

**Reply by zhiha** [reply]
Thank you for your response, I will give it a try ^_^
